src/qcodes/dataset/measurements.py

```diff
@@ -1278,23 +1288,37 @@ def unregister_parameter(self, parameter: setpoints_type) -> None:
         running this measurement
         """
         if isinstance(parameter, ParameterBase):
-            param = str_or_register_name(parameter)
+            param_name = str_or_register_name(parameter)
         elif isinstance(parameter, str):
-            param = parameter
+            param_name = parameter
         else:
             raise ValueError(
                 "Wrong input type. Must be a QCoDeS parameter or"
                 " the name (a string) of a parameter."
             )
 
         try:
-            paramspec: ParamSpecBase = self._interdeps[param]
+            paramspec: ParamSpecBase = self._interdeps[param_name]
         except KeyError:
             return
 
         self._interdeps = self._interdeps.remove(paramspec)
 
-        log.info(f"Removed {param} from Measurement.")
+        # Must follow interdeps removal, because interdeps removal may error
+        if isinstance(parameter, ParameterBase):
```

Comment:
This looks good, do you think it's worth adding a test?

Reply:
The existing tests cover this pretty well, it turns out. I had to fix a bunch of stuff to get them to pass.